.github/workflows/CI.yml

```diff
@@ -63,6 +63,8 @@ jobs:
       - uses: gradle/actions/setup-gradle@v3
         with:
           gradle-version: ${{ env.gradle-version }}
+          gradle-home-cache-cleanup: true
+          cache-read-only: false
```

Comment:
I don't think we want this setting enabled, that would allow non-main branches to push artifacts into the cache.  IMO I think only main should push artifacts
https://github.com/gradle/actions/blob/main/docs/setup-gradle.md#caching-build-state-between-jobs

Reply:
Removed `cache-read-only: false` and parallelism